src/runtime/auto-prompt-manager.js

```diff
@@ -365,56 +388,59 @@ export class AutoPromptManager {
    *   dismissedPrompts: (?string|undefined),
    *   shouldShowAutoPrompt: (boolean|undefined),
    * }} params
-   * @return {!string|undefined}
+   * @return {!Promise<string|undefined>}
    */
   getAudienceActionPromptType_({
     article,
     autoPromptType,
     dismissedPrompts,
     shouldShowAutoPrompt,
   }) {
-    let potentialActions = article?.audienceActions?.actions || [];
-    potentialActions = potentialActions.filter((action) =>
+    const audienceActions = article?.audienceActions?.actions || [];
+
+    const filterBooleans = audienceActions.map((action) =>
       this.checkActionEligibility_(action.type)
     );
+    return Promise.all(filterBooleans).then((booleans) => {
+      let potentialActions = audienceActions.filter((_, i) => booleans[i]);
```

Comment:
nit: I think it can be simplified as 
```
 return audienceActions.filter(...).then((potentialActions) => {
    ...
}
```

Reply:
I believe this requires the promises to resolve prior to the filtering action, as filtering on just the promises will always be truthy. Nonetheless, I have refactored this part, so marking resolved.